hearts/api/socket_events.py

```diff
@@ -66,3 +120,31 @@ def on_leave(data):
     room = data['room']
     io.leave_room(room)
     chat(username + ' has left the room.', room=room)
+
+
+@socketio.on('pass_cards')
+def on_pass_cards(data):
+    '''
+    data: {'cards': [str, str, str]}
+    '''
+    socket_id = request.sid
+    game_id = game_id_lookup()
+    game = get_game(game_id)
+    current_round = game.rounds[-1]
+    cards = [Card.deserialize(a) for a in data['cards']]
+    room = get_room(session['room'])
+
+    player = create_player(room, socket_id)
+    confirmation = {'status': 'success'}
+    try:
+        current_round.add_to_pass_selections(player, cards)
+        save_game(game, game_id)
+    except ValueError as error_message:
+        confirmation['status'] = 'failure'
+        confirmation['message'] = str(error_message)
+    finally:
+        io.emit('pass_submission_status', confirmation, room=socket_id)
```

Comment:
This finally block should just be a regular code block. In general, finally is used for cleanup, since it runs even in the event that an error is raised during the "except" block, or if a non ValueError is raised during the try. E.g., later we change some code in `add_to_pass_selections` that causes a divide by zero error, the result would be "success". We would want to see a failure in such a case.